samples/lucy_utils.cpp

```diff
@@ -0,0 +1,365 @@
+            .targetApi(filamat::MaterialBuilder::TargetApi::OPENGL)
+            .shading(Shading::UNLIT)
+            .depthWrite(false)
+            .depthCulling(false)
+            .build();
+        return Material::Builder().package(pkg.getData(), pkg.getSize()).build(engine);
+    }(*engine);
+
+    static const float4* weights = []() {
+        static float4 weights[FILTER_SIZE * 2];
+        float4* hweights = weights;
+        float4* vweights = weights + FILTER_SIZE;
+        static const float radius = 2;
+        const auto filter = [](float t) {
+            t /= 2.0;
+            if (t >= 1.0) return 0.0f;
+            const float scale = 1.0f / std::sqrt(0.5f * M_PIf);
+            return std::exp(-2.0f * t * t) * scale;
+        };
+        constexpr float pixelWidth = 2.0f / float(FILTER_SIZE);
+        float sum = 0.0f;
+        for (int i = 0; i < FILTER_SIZE; i++) {
+            float x = -1.0f + pixelWidth / 2.0f + pixelWidth * i;
+            hweights[i].x = vweights[i].x = filter(std::abs(x));
+            hweights[i].y = vweights[i].z = radius * (i - (FILTER_SIZE - 1) / 2);
+            hweights[i].z = vweights[i].y = 0.0f;
+            sum += weights[i].x;
+        }
+        for (int i = 0; i < FILTER_SIZE; i++) {
+            hweights[i].x /= sum;
+            vweights[i].x /= sum;
+        }
+        return weights;
+    }();
+    static const float4* hweights = weights;
+    static const float4* vweights = weights + FILTER_SIZE;
+
+    static Material* blur = [](Engine& engine) {
+        std::string txt = R"SHADER(
+            void material(inout MaterialInputs material) {
+                prepareMaterial(material);
+                float2 uv = gl_FragCoord.xy;
+                vec4 c = vec4(0);
+                for (int i = 0; i < FILTER_SIZE; i++) {
+                    float2 st = (uv + materialParams.weights[i].yz) * getResolution().zw;
+                    c += texture(materialParams_color, st) * materialParams.weights[i].x;
+                }
+                material.baseColor = c;
+            }
+        )SHADER";
+        const std::string from("FILTER_SIZE");
+        const std::string to = std::to_string(FILTER_SIZE);
+        size_t pos = txt.find(from);
+        txt.replace(pos, from.length(), to.c_str());
+        filamat::Package pkg = filamat::MaterialBuilder()
+            .name("hblur")
+            .material(txt.c_str())
+            .require(VertexAttribute::UV0)
+            .parameter(filamat::MaterialBuilder::SamplerType::SAMPLER_2D, "color")
+            .parameter(filamat::MaterialBuilder::UniformType::FLOAT4, FILTER_SIZE, "weights")
+            .targetApi(filamat::MaterialBuilder::TargetApi::OPENGL)
+            .shading(Shading::UNLIT)
+            .depthWrite(false)
+            .depthCulling(false)
+            .build();
+        return Material::Builder().package(pkg.getData(), pkg.getSize()).build(engine);
+    }(*engine);
+
+    TextureSampler::MinFilter minfilt;
```

Comment:
Nit: in this PR you have many variables that don't follow our naming convention. They should be camel case (`minFilt` not `minfilt`). Also avoid small abbreviations, use `minFilter`.

Reply:
Thanks, fixed this up.